ddtrace/contrib/botocore/services/bedrock.py

```diff
@@ -292,61 +262,63 @@ def _extract_streamed_response_metadata(span: Span, streamed_body: List[Dict[str
-    """Perform response param extraction and tagging."""
     metadata = result["ResponseMetadata"]
     http_headers = metadata["HTTPHeaders"]
-    span.set_tag_str("bedrock.response.id", str(metadata.get("RequestId", "")))
-    span.set_tag_str("bedrock.response.duration", str(http_headers.get("x-amzn-bedrock-invocation-latency", "")))
-    span.set_tag_str("bedrock.usage.prompt_tokens", str(http_headers.get("x-amzn-bedrock-input-token-count", "")))
-    span.set_tag_str("bedrock.usage.completion_tokens", str(http_headers.get("x-amzn-bedrock-output-token-count", "")))
 
-    # Wrap the StreamingResponse in a traced object so that we can tag response data as the user consumes it.
+    core.dispatch(
+        "botocore.patched_bedrock_api_call.success",
+        [
+            ctx,
+            str(metadata.get("RequestId", "")),
+            str(http_headers.get("x-amzn-bedrock-invocation-latency", "")),
+            str(http_headers.get("x-amzn-bedrock-input-token-count", "")),
+            str(http_headers.get("x-amzn-bedrock-output-token-count", "")),
+        ],
+    )
+
     body = result["body"]
-    result["body"] = TracedBotocoreStreamingBody(body, span, integration, prompt=prompt)
+    result["body"] = TracedBotocoreStreamingBody(body, ctx)
     return result
 
 
 def patched_bedrock_api_call(original_func, instance, args, kwargs, function_vars):
     params = function_vars.get("params")
-    trace_operation = function_vars.get("trace_operation")
-    operation = function_vars.get("operation")
     pin = function_vars.get("pin")
-    endpoint_name = function_vars.get("endpoint_name")
-    integration = function_vars.get("integration")
-    # This span will be finished once the user fully consumes the stream body, or on error.
-    bedrock_span = pin.tracer.trace(
-        trace_operation,
-        service=schematize_service_name("{}.{}".format(pin.service, endpoint_name)),
-        resource=operation,
+    model_provider, model_name = params.get("modelId").split(".")
+    with core.context_with_data(
+        "botocore.patched_bedrock_api_call",
+        pin=pin,
+        span_name=function_vars.get("trace_operation"),
+        service=schematize_service_name("{}.{}".format(pin.service, function_vars.get("endpoint_name"))),
+        resource=function_vars.get("operation"),
         span_type=SpanTypes.LLM,
-    )
-    prompt = None
-    try:
-        prompt = handle_bedrock_request(bedrock_span, integration, params)
-        result = original_func(*args, **kwargs)
-        result = handle_bedrock_response(bedrock_span, integration, result, prompt=prompt)
-        return result
-    except Exception:
-        _handle_exception(bedrock_span, integration, prompt, sys.exc_info())
-        raise
+        call_key="instrumented_bedrock_call",
+        call_trace=True,
+        bedrock_integration=function_vars.get("integration"),
+        params=params,
+        model_provider=model_provider,
+        model_name=model_name,
+    ) as ctx:
```

Comment:
For LLMObs spans, there are some things we like to do on span startup ([right now not much](https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/llmobs/_integrations/base.py#L121-L122), but in the near future we're planning on adding a couple temporary tags and processing). 

I was planning on implementing this by starting the span using `bedrock_integration.trace(...)` but obviously this won't be feasible if we move over to using `core.context_with_data()`. Do you know if we will be able to extract the same LLMObs-specific logic using the new core dispatch/handler framework?

Reply:
Yes, absolutely you can. Sounds like the approach I might take would be to have the LLMObs code listen to the context-started event fired from `context_with_data`, and start its own span in the event handler. I'd be happy to do a deeper review and/or pair with you on your code if you decide to take a similar approach.